pkg/common/contexts.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"context"
+	"sync"
+)
+
+func MergeContexts(ctxs ...context.Context) (context.Context, context.CancelFunc) {
```

Comment:
are we sure we _always_ want child contexts to be cancelled immediately if the parents is cancelled? Just want to make sure we don't cancel a DB transaction or something.

Reply:
We shouldn't use this for every time theres two contexts. This is mostly a helper for when merging two contexts is okay. For example, we're merging contexts before we call the LogStream.Stream function. In particular, we can break out of the for loop in handleStream() with just a single/wrapped context vs adding a global context to LogStream and then adding a second case statement in handleStream. 

We're also using the merged context in goroutines where we don't need to add a second case statement for a global context - at least I don't think we do. 1. because they're goroutines and errors aren't handled there and 2. I didn't see a reason to have different logic if the global context was cancelled.

Hopefully that makes sense. LMK if there's something that you thinking of specifically.

Reply:
Yep, that makes sense to me.